vendor/bat-native-ledger/src/bat_publishers.cc

```diff
@@ -204,14 +205,20 @@ void BatPublishers::saveVisitInternal(
   }
 
   std::string fav_icon = visit_data.favicon_url;
-  if (verified && fav_icon.length() > 0) {
+  if (verified && fav_icon.length() > 0 &&
+    fav_icon.find(".invalid") == std::string::npos) {
     ledger_->FetchFavIcon(fav_icon,
                           "https://" + ledger_->GenerateGUID() + ".invalid",
                           std::bind(&BatPublishers::onFetchFavIcon,
                                     this,
                                     publisher_info->id,
+                                    window_id,
                                     _1,
                                     _2));
+  } else {
+    if (verified) {
```

Comment:
This one needs to check for `fav_icon.length()` as well.

I think we need something like this
```
if (verified && fav_icon.length() > 0) {
  if (fav_icon.find(".invalid") == std::string::npos) {
   // fetch
  } else {
   // assign
  }
} else {
 publisher_info->favicon_url = "";
}

Added else at the end as we need to clear favicon if user goes from verified to unverified

Reply:
Fixed